### PR TITLE
Added multifield indexing with JSON collection

### PR DESCRIPTION
**Pull request by lintool**
This was implemented for MS MARCO doc2query experiments. Doesn't appear to help, but this code should be useful for other applications as well, so good to commit to repo.

### CODE REVIEW

File: src/main/java/io/anserini/search/SimpleSearcher.java [hunk 1]
```diff
@@ -208,6 +208,48 @@ public void close() throws IOException {
...
+  public Result[] searchFields(String q, String f, float boost, int k) throws IOException {
+    IndexSearcher searcher = new IndexSearcher(reader);
+    searcher.setSimilarity(similarity);
+
+    Query queryContents = new BagOfWordsQueryGenerator().buildQuery(LuceneDocumentGenerator.FIELD_BODY, analyzer, q);
+    Query queryField = new BagOfWordsQueryGenerator().buildQuery(f, analyzer, q);
+    BooleanQuery query = new BooleanQuery.Builder()
+        .add(queryContents, BooleanClause.Occur.SHOULD)
+        .add(new BoostQuery(queryField, boost), BooleanClause.Occur.SHOULD).build();
+
```
Comment (Victor0118) [inline src/main/java/io/anserini/search/SimpleSearcher.java]: So lines below are the same with those in the function `search()`, right? Packing them into one function would be better?

Comment (lintool) [reply src/main/java/io/anserini/search/SimpleSearcher.java]: Do you mean an overloaded method `search`?

File: src/test/java/io/anserini/collection/JsonDocumentObjectTest.java [hunk 2]
```diff
@@ -23,7 +23,12 @@
 import java.util.HashMap;
 import java.util.concurrent.atomic.AtomicInteger;
 
-
+// A file in a JsonCollection can either be:
+// (1) A single JSON object (i.e., a single document)
+// (2) An array of JSON objects
+// (3) JSON Lines (i.e., one JSON object per line)
```
Comment (Victor0118) [inline src/test/java/io/anserini/collection/JsonDocumentObjectTest.java]: IIUC, JsonCollection supports a line/lines of:
(1). JSON object
(2). JSON array

So it supports JSON array lines beside your list, right?

Comment (lintool) [reply src/test/java/io/anserini/collection/JsonDocumentObjectTest.java]: > So it supports JSON array lines beside your list, right?

What do you mean by this?

JSON Lines is JSONL: http://jsonlines.org/

Comment (Victor0118) [reply src/test/java/io/anserini/collection/JsonDocumentObjectTest.java]: I see. Then make sense. No need to modify.

File: src/main/java/io/anserini/search/SimpleSearcher.java (hunk 1 above)
Comment (Victor0118) [reply src/main/java/io/anserini/search/SimpleSearcher.java]: I mean wrapping line 225 to line 250 into one function and call it in both search() and searchFields().

**Pull request merged**